hw/mcu/native/src/hal_hw_id.c

```diff
@@ -22,20 +22,39 @@
 
 #include "hal/hal_bsp.h"
 
+#include "hal_native_priv.h"
+
 #ifndef min
 #define min(a, b) ((a)<(b)?(a):(b))
 #endif
 
+static uint8_t hal_hw_id[HAL_BSP_MAX_ID_LEN];
+static uint8_t hal_hw_id_len;
+
 /*
  * This can be used as the unique hardware identifier for the platform, as
  * it's supposed to be unique for this particular MCU.
  */
 int
 hal_bsp_hw_id(uint8_t *id, int max_len)
 {
+    if (hal_hw_id_len) {
+        memcpy(id, hal_hw_id, hal_hw_id_len);
```

Comment:
Should this function make sure it doesn't copy more than `max_len` bytes here?

Reply:
Yes, it should. Lemme fix that.